apps/chat/src/components/Common/ApplicationDialog.tsx

```diff
@@ -36,688 +33,561 @@ import Modal from '@/src/components/Common/Modal';
-  const [versionError, setVersionError] = useState<string | null>(null);
-  const [iconError] = useState<string | null>(null);
-  const nameInputRef = useRef<HTMLInputElement>(null);
-  const descriptionInputRef = useRef<HTMLTextAreaElement>(null);
-  const [filterParams, setFilterParams] = useState<string[]>([]);
-  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
-  const [isPublishing, setIsPublishing] = useState(false);
-
-  const entity = selectedApplication && {
-    name: selectedApplication.name,
-    id: ApiUtils.decodeApiUrl(
-      selectedApplication.id ||
-        (selectedApplication as unknown as { application: string }).application,
-    ),
-    folderId: getFolderIdFromEntityId(selectedApplication.name),
-  };
-
-  const resetForm = () => {
-    setName('');
-    setVersion('');
-    setDescription('');
-    setFeatures('');
-    setMaxAttachments(0);
-    setCompletionUrl('');
-    setLocalLogoFile(undefined);
-    setDeleteLogo(false);
-    setFilterParams([]);
-  };
+}) => {
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    clearErrors,
+    trigger,
+    control,
+    formState: { errors, isValid },
+  } = useForm<FormData>({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  });
 
-  const loading = useAppSelector((state) => state.application.loading);
+  const { t } = useTranslation(Translation.Chat);
 
-  const handleChangeFilterParams = useCallback((filterParams: string[]) => {
-    setFilterParams(filterParams);
-  }, []);
+  const dispatch = useAppDispatch();
 
-  const handleClose = useCallback(() => {
-    setSubmitted(false);
-    resetForm();
-    onClose(false);
-  }, [onClose]);
+  const selectedApplication = useAppSelector(
+    ApplicationSelectors.selectApplicationDetail,
+  );
+  const files = useAppSelector(FilesSelectors.selectFiles);
 
-  const nameOnBlurHandler = (e: FocusEvent<HTMLInputElement>) => {
-    const newName = prepareEntityName(e.target.value.trim(), {
-      forRenaming: true,
-    });
-    const pattern = /^[^!@#$^*()+]{2,160}$/;
-
-    if (!pattern.test(newName)) {
-      setNameError(
-        t(
-          'Name should be 2 to 160 characters long and should not contain special characters',
-        ),
-      );
-    } else {
-      setNameError(null);
-    }
-    setName(newName);
-    onBlur(e);
-  };
+  const selectedApplicationId = getSelectedApplicationId(selectedApplication);
 
-  const nameOnChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
-    const newName = e.target.value;
-    setName(newName);
+  const [deleteLogo, setDeleteLogo] = useState(false);
+  const [localLogoFile, setLocalLogoFile] = useState<string | undefined>();
+  const [inputAttachmentTypes, setInputAttachmentTypes] = useState<string[]>(
+    [],
+  );
+  const [featuresInput, setFeaturesInput] = useState(
+    isEdit && selectedApplication && selectedApplication.features
+      ? safeStringify(selectedApplication.features)
+      : '',
+  );
```

Comment:
a lot of places with `isEdit && selectedApplication`
Why can we just use something like 
```suggestion
  const [featuresInput, setFeaturesInput] = useState(safeStringify(selectedApplication?.features));
```
everywhere and read `selectedApplication` only if `isEdit` (see https://github.com/epam/ai-dial-chat/pull/1965#discussion_r1733448679)?